src/Controller/CarteController.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Category;
+use App\Repository\CategoryRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class CarteController extends AbstractController
+{
+    /**
+     * @Route("/carte", name="carte")
+     */
+    public function index(): Response
+    {
+        /** @var CategoryRepository $repository */
+        $repository = $this->getDoctrine()->getRepository(Category::class);
+        $categories = $repository->findWithProducts();
```

Comment:
why did you use this function? "Finby" would have been enough? no?

Reply:
no because i'm using left join in this function